levels/level_manager.go

```diff
@@ -371,21 +372,21 @@ func (lm *LevelManager) getOverlappingTables(keyStart []byte, keyEnd []byte, lev
 
 const maxRetries = 100
 
-func (lm *LevelManager) QueryTablesInRange(keyStart []byte, keyEnd []byte) (OverlappingTables, error) {
+func (lm *LevelManager) QueryTablesInRange(keyStart []byte, keyEnd []byte) (QueryTablesResult, error) {
 	if lm.getState() != stateActive {
-		return nil, common.NewTektiteErrorf(common.Unavailable, "levelManager not active")
+		return QueryTablesResult{}, common.NewTektiteErrorf(common.Unavailable, "levelManager not active")
 	}
 	retryCount := 0
 outer:
 	for {
 		// We load the readable state from an atomic pointer, this state is updated atomically after apply changes have
 		// been applied to the LSM, this allows us to reduce contention between the read path and the write path.
 		readable := lm.readable.Load()
```

Comment:
The idea behind readable state is we use a copy on write approach. The readable state is set when it's updated, and then query can read the last readable state without having to use a lock, which reduces contention.
This means the query results returned all must be built from the readable state, it's not correct to use another lock and get the L0 groups, as the current writable state won't correspond to the same state if an update has been done in between.